hbase-server/src/main/java/org/apache/hadoop/hbase/quotas/QuotaCache.java

```diff
@@ -140,8 +150,7 @@ public QuotaLimiter getUserLimiter(final UserGroupInformation ugi, final TableNa
    */
   public UserQuotaState getUserQuotaState(final UserGroupInformation ugi) {
     return computeIfAbsent(userQuotaCache, getQuotaUserName(ugi),
-      () -> QuotaUtil.buildDefaultUserQuotaState(rsServices.getConfiguration(), 0L),
-      this::triggerCacheRefresh);
+      () -> QuotaUtil.buildDefaultUserQuotaState(rsServices.getConfiguration(), 0L));
```

Comment:
The change here, and in getQuotaState, are critical. These changes ensure that each cache miss does not trigger an immediate refresh — particularly given that cache entries are evicted after 5 refresh periods, this approach is too heavy handed.